source/routing/tests/routing_api_test.cpp

```diff
@@ -1560,6 +1561,81 @@ BOOST_FIXTURE_TEST_CASE(biking_length_test, streetnetworkmode_fixture<normal_spe
     BOOST_CHECK_EQUAL(section.duration(), total_dur);
 }
 
+/**
+  * Test length with a different speed factors
+ */
+
+BOOST_FIXTURE_TEST_CASE(speed_factor_length_test, streetnetworkmode_fixture<normal_speed_provider>) {
+    auto mode = navitia::type::Mode_e::Walking;
+    // you're supposed to be able to put any value here (only rounding might break tests)
+    double speed_factor = 1.5;
+    origin.streetnetwork_params.mode = mode;
+    origin.streetnetwork_params.offset = 0;
+    origin.streetnetwork_params.max_duration = bt::pos_infin;
+    origin.streetnetwork_params.speed_factor = speed_factor;
+
+    destination.streetnetwork_params.mode = mode;
+    destination.streetnetwork_params.offset = 0;
+    destination.streetnetwork_params.max_duration = bt::pos_infin;
+    destination.streetnetwork_params.speed_factor = speed_factor;
+
+    pbnavitia::Response resp = make_response();
+
+    pbnavitia::Journey journey;
+    for (int idx = 0; idx < resp.journeys().size(); ++idx) {
+        if (resp.journeys(idx).sections().size() == 1) {
+            journey = resp.journeys(idx);
+        }
+    }
+
+    BOOST_REQUIRE_EQUAL(journey.sections_size(), 1);
+    pbnavitia::Section section = journey.sections(0);
+    BOOST_REQUIRE_EQUAL(section.type(), pbnavitia::SectionType::STREET_NETWORK);
+
+    dump_response(resp, "speed factor length test");
```

Comment:
You have to set it explicitly to `true` if you want the dump.
So useless code, but no useless trace. The point is to ease the debug, as on the other tests around.